### ISSUE TITLE
Changes to docker-compose.yml and run.sh

### ISSUE THREAD

**Issue by frontendjim** [visiblevc/wordpress-starter#120]
## Overview

- **What's your problem about?** 
I'm trying to add a couple of changes to the run.sh file and docker-compose.yml file to have dedicated environment variables for things like the admin user and admin password and title to avoid exposing database credentials, etc but no matter when I try, my changes don't seem to make any difference and I can still log in to wordpress with root/root and the title is the same as database. Any ideas why this might be?

- **What is your operating system?** 
`OSX 10.13.1`

- **What is your docker version?** 
`17.09.0-ce, build afdb6d4`

## `docker-compose.yml`

```
version: '3'
services:
  wordpress:
    image: visiblevc/wordpress:0.16.0-php7.1
    ports:
      - 8080:80
      - 443:443
    volumes:
      - ./data:/data
      - ./wp-content:/app/wp-content
    environment:
      DB_NAME: csCareers
      DB_PASS: root
      WP_VERSION: 4.9
      ADMIN_USER: admin
      ADMIN_PASSWORD: 047!1ZueyRL8
      TITLE: Careers
  db:
    image: mariadb:10 # or mysql:5.7
    volumes:
      - data:/var/lib/mysql
    environment:
      MYSQL_ROOT_PASSWORD: root
volumes:
  data:
```

## `run.sh`
```
# Environment
# ------------
DB_HOST=${DB_HOST:-db}
DB_NAME=${DB_NAME:-wordpress}
DB_PASS=${DB_PASS:-root}
DB_PREFIX=${DB_PREFIX:-wp_}
PERMALINKS=${PERMALINKS:-'/%year%/%monthnum%/%postname%/'}
SERVER_NAME=${SERVER_NAME:-localhost}
WP_VERSION=${WP_VERSION:-latest}
# FIXME: Remove in next version
URL_REPLACE=${URL_REPLACE:-"$SEARCH_REPLACE"}
BEFORE_URL="${URL_REPLACE%,*}"
AFTER_URL="${URL_REPLACE#*,}"
# WORDPRESS: Config options
TITLE=${TITLE:-wordpress}
ADMIN_EMAIL=${ADMIN_EMAIL:-test@test.com}
ADMIN_USER=${ADMIN_USER:-root}
ADMIN_PASSWORD=${ADMIN_PASSWORD:-root}

declare -A plugin_deps
declare -A theme_deps
declare -A plugin_volumes
declare -A theme_volumes

# Apache configuration
# --------------------
sed -i "s/#ServerName www.example.com/ServerName $SERVER_NAME\nServerAlias www.$SERVER_NAME/" /etc/apache2/sites-available/000-default.conf

# WP-CLI configuration
# ---------------------
cat > ~/.wp-cli/config.yml <<EOF
apache_modules:
  - mod_rewrite

core config:
  dbuser: root
  dbpass: $DB_PASS
  dbname: $DB_NAME
  dbprefix: $DB_PREFIX
  dbhost: $DB_HOST:3306
  extra-php: |
    define('WP_DEBUG', ${WP_DEBUG:-false});
    define('WP_DEBUG_LOG', ${WP_DEBUG_LOG:-false});
    define('WP_DEBUG_DISPLAY', ${WP_DEBUG_DISPLAY:-true});

core install:
  url: ${AFTER_URL:-localhost:8080}
  title: $TITLE
  admin_user: $ADMIN_USER
  admin_password: $ADMIN_PASSWORD
  admin_email: $ADMIN_EMAIL
  skip-email: true
EOF
```

## Project structure

```
/data
  /database.sql
/wp-content
  /plugins
    /my-plugin
  /themes
    /my-theme
```

## `docker-compose up` output

```
Starting example_db_1 ...
Starting example_db_1
Starting example_wordpress_1 ...
Starting example_wordpress_1 ... done
Attaching to example_db_1, example_wordpress_1
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] mysqld (mysqld 10.2.10-MariaDB-10.2.10+maria~jessie) starting as process 1 ...
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Uses event mutexes
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Compressed tables use zlib 1.2.8
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Using Linux native AIO
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Number of pools: 1
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Using SSE2 crc32 instructions
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Initializing buffer pool, total size = 256M, instances = 1, chunk size = 128M
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Completed initialization of buffer pool
db_1         | 2017-11-28 14:28:16 140080019363584 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Highest supported file format is Barracuda.
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: 128 out of 128 rollback segments are active.
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Creating shared tablespace for temporary tables
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: Waiting for purge to start
wordpress_1  | ===============================================================================
wordpress_1  |                          Begin WordPress Installation
wordpress_1  | ===============================================================================
db_1         | 2017-11-28 14:28:16 140080754984832 [Note] InnoDB: 5.7.20 started; log sequence number 4984771
db_1         | 2017-11-28 14:28:16 140079855023872 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
db_1         | 2017-11-28 14:28:16 140079855023872 [Note] InnoDB: Buffer pool(s) load completed at 171128 14:28:16
db_1         | 2017-11-28 14:28:17 140080754984832 [Note] Plugin 'FEEDBACK' is disabled.
db_1         | 2017-11-28 14:28:17 140080754984832 [Note] Server socket created on IP: '::'.
db_1         | 2017-11-28 14:28:17 140080754984832 [Warning] 'proxies_priv' entry '@% root@c10db46b974a' ignored in --skip-name-resolve mode.
db_1         | 2017-11-28 14:28:17 140080754984832 [Note] Reading of all Master_info entries succeded
db_1         | 2017-11-28 14:28:17 140080754984832 [Note] Added new Master_info '' to hash table
db_1         | 2017-11-28 14:28:17 140080754984832 [Note] mysqld: ready for connections.
db_1         | Version: '10.2.10-MariaDB-10.2.10+maria~jessie'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
wordpress_1  | ==> Waiting for MySQL to initialize...
wordpress_1  | mysqld is alive
wordpress_1  | ==> Configuring WordPress
wordpress_1  | Success: Generated 'wp-config.php' file.
wordpress_1  | Success: core config
wordpress_1  | ==> Checking database
wordpress_1  | ==> Checking themes
wordpress_1  | ==> Checking plugins
wordpress_1  | Success: Plugin already activated.
wordpress_1  | Success: plugin activate all
wordpress_1  | ==> Finalizing
wordpress_1  | Success: Rewrite rules flushed.
wordpress_1  | Success: Rewrite structure set.
wordpress_1  | Success: rewrite structure
wordpress_1  | Success: Rewrite rules flushed.
wordpress_1  | Success: rewrite flush
wordpress_1  | ===============================================================================
wordpress_1  |                        WordPress Configuration Complete!
wordpress_1  | ===============================================================================
wordpress_1  | [Tue Nov 28 14:28:22.823361 2017] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.10 (Debian) PHP/7.1.9 configured -- resuming normal operations
wordpress_1  | [Tue Nov 28 14:28:22.823399 2017] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
```

**Comment by dsifford** [comment]
Changes to `run.sh` will only be reflected if you build your own image. The reason why they currently are not being reflected is because you are pulling the image we built from the docker registry and that image uses our version of `run.sh`

I'm not sure I understand what you're trying to do though. Can you clarify?

You said you're trying to "have dedicated environment variables for things like the admin user and admin password and title to avoid exposing database credentials", but in your example, you're not doing much other than creating two new environment variables that are equally as exposed as the predefined ones.

If security is your concern, you might want to look into using [docker secrets](https://docs.docker.com/engine/swarm/secrets/) (if building with swarm) or [pulling your environment variables from a `.env` file](https://docs.docker.com/compose/environment-variables/#the-env-file).

**TL;DR:** You need to build your own image. You can do that using the [`build`](https://docs.docker.com/compose/compose-file/#build) instruction, rather than the `image` instruction in your compose file. 

Of note: I've got some plans to make the current configuration a bit easier to extend without needing to build a custom image, but I'm not sure when those plans will land just yet. Should be soon(ish) if time allows.

**Comment by frontendjim** [comment]
Hi @dsifford 

Thank you so much for coming back to me. You're totally right, I needed to rebuild the image... d'oh! 

Thanks again.
